File: src/protocols/Protocols.Core/Protocol.cs
using SharedKernel;
using System;
using System.Collections.Generic;

namespace Protocols.Core
{
    public class Protocol : Entity
    {
        public string Name { get; set; }
		public string Description { get; set; }
		public DateTime? EnrollmentWindow { get; set;}
		public IList<Visit> Visits { get; set; }
    }
}

File: src/shell/assets/js/protocolApp/models/protocol.js
ProtocolApp.Protocol = DS.Model.extend({
  name: DS.attr("string"),
  description: DS.attr("string"),
  enrollmentWindow: DS.attr('string'),
  //inclusions:,
  //exclusions:,
  visits: DS.hasMany('visit', {inverse: null, async: true}),
  procedures: DS.hasMany('procedure', {inverse: null, async: true}),
  //unscheduledVisits: DS.hasMany('visit'),
});

/*ProtocolApp.ProtocolSerializer = DS.RESTSerializer.extend(DS.EmbeddedRecordsMixin, DS.NoKeyMixin, {
  attrs: {
    visits: { embedded: 'always', noKey: true },
    procedures: { embedded: 'always', noKey: true },
  }
});*/
